src/lambda_vm/interpreter.ml

```diff
@@ -0,0 +1,192 @@
+open Ir
+open Checks
+
+type error =
+  (* interpreter bugs *)
+  | Undefined_variable
+  | Over_applied_primitives
+  (* user program bugs *)
+  | Value_is_not_pair
+  | Value_is_not_int64
+  | Value_is_not_function
+  | Value_is_not_zero
+
+exception Error of error
+let raise error = raise (Error error)
+
+module Pattern : sig
```

Comment:
OCaml newbie question: Why is this part not in the `.mli` file?

Reply:
Because this is not used externally, it's an internal module of this module.